GeoClient/GeoClient/Services/Registration/RegistrationService.cs

```diff
@@ -85,6 +86,21 @@ public RegistrationInfo GetRegistrationInfo()
             return _cachedRegistrationInfo;
         }
 
+        public bool OpenUri()
+        {
+            bool result = false;
+            if (IsRegistered())
+            {
+                Device.OpenUri(new Uri(_cachedRegistrationInfo.Url));

```

Comment:
The service should not know something about UI concerns. I will move this to the UI class

Reply:
Yes, please. I don't have the build env and chose the least risk ...